presto-native-execution/presto_cpp/main/QueryContextManager.cpp

```diff
@@ -141,6 +145,15 @@ void updateVeloxConfigs(
     configStrings.emplace(
         core::QueryConfig::kDriverCpuTimeSliceLimitMs, "1000");
   }
+
+  for (auto& entry : connectorConfigStrings) {
+    auto connectorConfig = entry.second;
+    it = connectorConfig.find("node_selection_strategy");
+    if (it == connectorConfig.end() || it->second != "SOFT_AFFINITY") {
+      connectorConfig.emplace(
```

Comment:
The default value for Presto java is true? Thanks!

Reply:
In Java, there are three node selection strategy:  _HARD_AFFINITY_, _SOFT_AFFINITY_ and _NO_PREFERENCE_. _NO_PREFERENCE_ is the default value.

Code pointers shared by @kewang1024 : Splits are set as `cacheable` only when the strategy is _HARD_AFFINITY_ or _SOFT_AFFINITY_.
https://github.com/prestodb/presto/blob/6e4896ddc4ed078cce1f81186e6986ecca94a141/presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java#L156-L167

And _HARD_AFFINITY_ is not used anymore, so we expect cache is enabled only when soft affinity is set and the default is without cache.

Reply:
> And _HARD_AFFINITY_ is not used anymore, so we expect cache is enabled only when soft affinity is set and the default is without cache.

If _HARD_AFFINITY_ is not used anymore, should it be removed from the Presto doc in https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/develop/connectors.rst#connectorsplit ?

Reply:
HARD_AFFINITY is designed for Raptor, which is not used anymore by Meta.
But open source could still use it, so we can not remove it.

Reply:
Raptor was deleted last year by #19236

Reply:
Thanks @kewang1024 and @tdcmeehan. If Raptor was removed from OSS Presto over a year ago, is there any other use of _HARD_AFFINITY_ in current Presto that it should stay in the current doc?

Reply:
I found other `HARD_AFFINITY` usage for example `LocalFileSplit`

Reply:
> I found other `HARD_AFFINITY` usage for example `LocalFileSplit`

Thanks @kewang1024! In that case I think it should stay in the current doc.

Reply:
@zacw7 then change to it->second == "NO_PREFERENCE"? Also make sure it doesn't break existing meta native use case with SSD cache enabled? thanks!